js/src/forum/states/PostStreamState.js

```diff
@@ -133,6 +134,7 @@ class PostStreamState {
     this.needsScroll = true;
     this.targetPost = { index };
     this.animateScroll = !noAnimation;
+    this.previousIndex = index === this.previousIndex ? index : this.index;
```

Comment:
I wonder if we might want to also support this on goToNumber, so that back position is retained when clicking on a mention?